macrosynergy/management/utils/sparse.py

```diff
@@ -0,0 +1,821 @@
+            return getattr(cls, item)
+        else:
+            raise KeyError(f"{item} is not a valid method name")
+
+
+class StandardDeviationMethods(metaclass=SubscriptableMeta):
+    """
+    Class to hold methods for calculating standard deviations.
+    Each method must comply to the following signature:
+        `func(s: pd.Series, **kwargs) -> pd.Series`
+
+    Currently supported methods are:
+
+    - `std`: Standard deviation
+    - `abs`: Absolute standard deviation
+    - `exp`: Exponentially weighted standard deviation
+    - `exp_abs`: Exponentially weighted absolute standard deviation
+    """
+
+    @staticmethod
+    def std(s: pd.Series, min_periods: int, **kwargs) -> pd.Series:
+        """
+        Calculate the standard deviation of a Series.
+
+        :param <pd.Series> s: The Series to calculate the standard deviation for.
+        :param <int> min_periods: The minimum number of periods required for the calculation.
+        :return: The standard deviation of the Series.
+        """
+        return s.expanding(min_periods=min_periods).std()
+
+    @staticmethod
+    def abs(s: pd.Series, min_periods: int, **kwargs) -> pd.Series:
+        """
+        Calculate the absolute standard deviation of a Series.
+
+        :param <pd.Series> s: The Series to calculate the absolute standard deviation for.
+        :param <int> min_periods: The minimum number of periods required for the calculation.
+        :return: The absolute standard deviation of the Series.
+        """
+        return s.abs().expanding(min_periods=min_periods).mean()
+
+    @staticmethod
+    def exp(s: pd.Series, halflife: int, min_periods: int, **kwargs) -> pd.Series:
+        """
+        Calculate the exponentially weighted standard deviation of a Series.
+
+        :param <pd.Series> s: The Series to calculate the exponentially weighted standard
+            deviation for.
+        :param <int> halflife: The halflife of the exponential weighting.
+        :param <int> min_periods: The minimum number of periods required for the
+            calculation.
+        :return: The exponentially weighted standard deviation of the Series.
+        """
+        return s.ewm(halflife=halflife, min_periods=min_periods).std()
+
+    @staticmethod
+    def exp_abs(s: pd.Series, halflife: int, min_periods: int, **kwargs) -> pd.Series:
+        """
+        Calculate the exponentially weighted absolute standard deviation of a Series.
+
+        :param <pd.Series> s: The Series to calculate the exponentially weighted absolute
+            standard deviation for.
+        :param <int> halflife: The halflife of the exponential weighting.
+        :param <int> min_periods: The minimum number of periods required for the calculation.
+        :return: The exponentially weighted absolute standard deviation of the Series.
+        """
+        return s.abs().ewm(halflife=halflife, min_periods=min_periods).mean()
```

Comment:
Please correct me if I'm wrong, but this also looks like it's calculating the mean instead of std.